velox/connectors/hive/iceberg/IcebergPartitionIdGenerator.h

```diff
@@ -0,0 +1,127 @@
+  /// in the input RowVector.
+  /// @param maxPartitions Maximum number of distinct partitions allowed.
+  /// Exceeding this limit will cause a VeloxUserError.
+  /// @param pool Memory pool for allocating vectors and buffers.
+  /// @param columnTransforms Transforms to apply to each partition column.
+  /// Must have same size as partitionChannels.
+  IcebergPartitionIdGenerator(
+      std::vector<column_index_t> partitionChannels,
+      uint32_t maxPartitions,
+      memory::MemoryPool* pool,
+      const std::vector<TransformPtr>& columnTransforms);
+
+  /// Generate sequential partition IDs for all rows in the input vector.
+  ///
+  /// For each row, applies the partition transforms, hashes the transformed
+  /// values, and maps to a sequential partition ID. New partitions are created
+  /// as needed (up to maxPartitions limit). Partition values are stored for
+  /// later partition name generation.
+  ///
+  /// @param input Input RowVector containing all columns. Must have columns
+  /// at indices specified by partitionChannels.
+  /// @param result Output vector of partition IDs, one per input row.
+  /// Resized to match input size. Values are 0-based sequential IDs.
+  void run(const RowVectorPtr& input, raw_vector<uint64_t>& result) override;
+
+  /// Generate the partition name for the given partition ID.
+  ///
+  /// Construct an Iceberg-compliant partition path string in the format:
+  /// "key1=value1/key2=value2/..." where keys are partition column names for
+  /// identity transform or column_transform for non-identity transforms and
+  /// values are the human-readable string representations of the transformed
+  /// partition values. Special characters are URL-encoded per
+  /// java.net.URLEncoder.encode().
+  ///
+  /// @param partitionId Sequential partition ID (0-based) returned by run().
+  /// @return Partition path string applicable for use in file paths.
+  std::string partitionName(uint32_t partitionId) const override;
+
+  /// Return the RowVector containing transformed partition values.
+  /// Each row in this vector corresponds to a partition ID (row index =
+  /// partition ID). Each column contains the transformed values for that
+  /// partition column.
+  ///
+  /// @return RowVector with one column per partition, columns in same order as
+  /// columnTransforms. Returns nullptr if no partitions have been created.
+  RowVectorPtr partitionValues() const {
+    return partitionValues_;
+  }
+
+  /// Convert a partition value to its string representation for use in
+  /// partition directory path. The format follows the Iceberg specification
+  /// for partition path encoding.
+  ///
+  /// @param value The partition value to convert.
+  /// @param transform The transform that produced this partition value.
+  /// @return String representation applicable for partition directory names.
+  template <typename T>
+  std::string partitionValueToString(T value, const TransformPtr& transform)
```

Comment:
Does partitionValueToString need to be a method of the class? It seems like it could be a free function in the .cpp file as it doesn't use any member variables.